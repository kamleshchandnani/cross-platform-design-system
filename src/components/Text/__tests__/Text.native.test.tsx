import React from 'react';
import { RenderText } from './RenderText';
import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';

beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
afterAll(() => jest.restoreAllMocks());

describe('<Text />', () => {
  it('should render regular text', () => {
    const displayText = 'Displaying some text';
    const { toJSON, getByText } = renderWithTheme(<RenderText>{displayText}</RenderText>);
    expect(getByText('Displaying some text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
