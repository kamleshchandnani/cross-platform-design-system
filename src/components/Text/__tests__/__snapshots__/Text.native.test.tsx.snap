// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render regular text 1`] = `
<Text
  color="surface.text.normal.highContrast"
  fontFamily="text"
  fontSize={25}
  fontWeight="regular"
  lineHeight="m"
  style={
    Array [
      Object {
        "color": "hsla(0, 0%, 100%, 1)",
        "fontFamily": "Lato",
        "fontSize": 11,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 18,
        "marginBottom": 16,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "textAlign": "left",
        "textDecorationLine": "none",
      },
    ]
  }
>
  Displaying some text
</Text>
`;
